File: 202102660/10845.cc
#include <iostream>
#include <queue>
using namespace std;
int main() {
    int N;
    string inst;
    queue<int> q;
    cin >> N;

    for (int i=0; i <N; i++) {
        cin >> inst;
        if (inst.compare("push") == 0) {
            int x;
            cin >> x;
            q.push(x);
        } else if (inst.compare("pop") == 0) {
            if (q.empty()) {
                cout << -1 << '\n';
            } else {
                cout << q.front() << '\n';
                q.pop();
            }
        } else if (inst.compare("size") == 0) {
            cout << q.size() << '\n';
        } else if (inst.compare("empty") == 0) {
            if (q.empty()) {
                cout << 1 << '\n';
            } else {
                cout << 0 << '\n';
            }
        } else if (inst.compare("front") == 0) {
            if (q.empty()) {
                cout << -1 << '\n';
            } else {
                cout << q.front() << '\n';
            }
        } else if (inst.compare("back") == 0) {
            if (q.empty()) {
                cout << -1 << '\n';
            } else {
                cout << q.back() << '\n';
            }
        } 
    }
}

File: 201902693/10845.py
import sys
input = sys.stdin.readline
n = int(input())
queue = []
for i in range(n) :
    cmd = list(input().split())
    if cmd[0] == 'push' :
        queue.append(int(cmd[1]))
    if cmd[0] == 'pop' :
        if len(queue) == 0 :
            print(-1)
        else :
            print(queue[0])
            del queue[0]
    if cmd[0] == 'size' :
        print(len(queue))
    if cmd[0] == 'empty' :
        if len(queue) == 0 :
            print(1)
        else :
            print(0)
    if cmd[0] == 'front' :
        if len(queue) == 0 :
            print(-1)
        else :
            print(queue[0])
    if cmd[0] == 'back' :
        if len(queue) == 0 :
            print(-1)
        else :
            print(queue[-1])